code/modules/mob/mob_grab_specials.dm

```diff
@@ -147,3 +147,46 @@
 
 	if(can_eat)
 		H.attempt_devour(target, H.eat_types, H.mouth_size)
+
+
+/obj/item/weapon/grab/proc/hair_pull(mob/living/carbon/human/target, mob/attacker, var/target_zone)
+
+
+	var/mob/living/carbon/human/H = target
+	var/obj/item/weapon/grab/G = attacker.get_active_hand()
+	var/datum/sprite_accessory/hair/hair_style = hair_styles_list[H.h_style]
+	var/hairchatname = hair_style.chatname
+	for(var/obj/item/protection in list(H.head, H.wear_mask))
+		if(protection && (protection.body_parts_covered & HEAD))
+			assailant << "<span class='danger'>You can't grab their hair while they wear a head cover.</span>"

```

Comment:
This language is pretty inelegant. Try "You can't tug their hair while something is covering it!"